app/src/main/java/com/locoquest/app/MainActivity.kt

```diff
@@ -32,14 +32,12 @@ class MainActivity : AppCompatActivity(), OnMapReadyCallback{
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
-
         when (requestCode) {
             REQ_ONE_TAP -> {
                 try {
                     val credential = oneTapClient.getSignInCredentialFromIntent(data)
                     val idToken = credential.googleIdToken
-                    when {
```

Comment:
Nested when structures are difficult to understand because you can easily confuse the cases of an inner when as belonging to an outer statement. Therefore nested when statements should be avoided.
(https://rules.sonarsource.com/kotlin/RSPEC-1821)